compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangPackage.java

```diff
@@ -99,7 +100,7 @@ public BLangPackage() {
         this.completedPhases = EnumSet.noneOf(CompilerPhase.class);
         this.testablePkgs = new ArrayList<>();
         this.flagSet = EnumSet.noneOf(Flag.class);
-        this.diagnostics = new ArrayList<>();
+        this.diagnostics = new TreeSet<>();
```

Comment:
Let use the constructor that accepts a comparator. Then we can avoid the diagnostic related classes from having to implement `Comparable<T>`